File: 0414-third-maximum-number/0414-third-maximum-number.cpp
class Solution{
public:
    int thirdMax(vector<int>&n){
        set<int>s(n.begin(),n.end());
        vector<int>v(s.begin(),s.end());
        reverse(v.begin(),v.end());
        return v.size()<3?v[0]:v[2];
    }
};

File: 0414-third-maximum-number/0414-third-maximum-number.py
class Solution:
    def thirdMax(self, nums: List[int]) -> int:
        first = second = third = None
        for x in nums:
            if x == first or x == second or x == third:
                continue
            if first is None or x > first:
                third = second
                second = first
                first = x
            elif second is None or x > second:
                third = second
                second = x
            elif third is None or x > third:
                third = x
        return first if third is None else third
